src/hotspot/share/nmt/vmatree.cpp

```diff
@@ -77,6 +77,7 @@ VMATree::SummaryDiff VMATree::register_mapping(position A, position B, StateType
       assert(leqA_n->val().out.type() != StateType::Released, "Should not use inplace the tag of a released region");
       MemTag tag = leqA_n->val().out.mem_tag();
       stA.out.set_tag(tag);
+      LEQ_A.state.out.set_tag(tag);
```

Comment:
Shouldn't `LEQ_A.state.out`'s tag already be `tag`? Is this line really necessary?

Reply:
It's not. Removed.